### PR TITLE
(maint) Add initial set of wix vars

### PR DESCRIPTION
**Pull request by melissa**
This commit adds in a base set of wix variables for processing templates

### CODE REVIEW

File: configs/projects/puppet-agent.rb [hunk 1]
```diff
@@ -98,6 +98,36 @@
     proj.identifier "com.puppetlabs"
   end
 
+  # WIX settings
+  if platform.is_windows?
+    proj.setting(:company_name, "Puppet Labs")
+    proj.setting(:company_word, "PuppetLabs")
+    proj.setting(:common_product_name, "PuppetInstaller")
+    proj.setting(:upgrade_code, "2AD3D11C-61B3-4710-B106-B4FDEC5FA358")
```
Comment (stahnma) [inline configs/projects/puppet-agent.rb]: Where does this come from?

Comment (melissa) [inline configs/projects/puppet-agent.rb]: https://github.com/puppetlabs/puppet_for_the_win/blob/master/wix/include/puppet.wxi#L35

File: configs/projects/puppet-agent.rb [hunk 2]
```diff
@@ -2,19 +2,46 @@
...
+    proj.setting(:UI_exitdialogtext, "Manage your first resources on this node, explore the Puppet community and get support using the shortcuts in the Documentation folder of your Start Menu.")
+
+    # Directory IDs
+    proj.setting(:bindir_id, "bindir")
+
+    # We build for windows not in the final destination, but in the paths that correspond
+    # to the directory ids expected by WIX. This will allow for a portable installation (ideally).
+    proj.setting(:install_root, File.join("C:", proj.base_dir, proj.company_id))
+    proj.setting(:prefix, File.join(proj.install_root, proj.product_id))
+    proj.setting(:sysconfdir, File.join("C:", "CommonAppDataFolder", proj.company_id))
```
Comment (melissa) [inline configs/projects/puppet-agent.rb]: `<Directory Id="CommonAppDataFolder" Name="CommonAppData">`

@jpogran can this be like `Program Files` where we only specify the ID?

File: configs/projects/puppet-agent.rb [hunk 3]
```diff
@@ -2,19 +2,46 @@
...
+        :ManualLink => "http://links.puppetlabs.com/windows-manual-foss",
+      })
+    proj.setting(:UI_exitdialogtext, "Manage your first resources on this node, explore the Puppet community and get support using the shortcuts in the Documentation folder of your Start Menu.")
+
+    # Directory IDs
+    proj.setting(:bindir_id, "bindir")
+
+    # We build for windows not in the final destination, but in the paths that correspond
+    # to the directory ids expected by WIX. This will allow for a portable installation (ideally).
+    proj.setting(:install_root, File.join("C:", proj.base_dir, proj.company_id))
```
Comment (melissa) [inline configs/projects/puppet-agent.rb]: This should end with `company_name`, but I can't get it to build with spaces still

File: configs/projects/puppet-agent.rb [hunk 4]
```diff
@@ -180,11 +206,19 @@
     proj.component "ruby-selinux"
   end
 
-  proj.directory proj.install_root
-  proj.directory proj.prefix
-  proj.directory proj.sysconfdir
+
+  if platform.is_windows?
+    proj.directory proj.install_root, id: proj.company_id, name: proj.company_name
```
Comment (MikaelSmith) [inline configs/projects/puppet-agent.rb]: These look like new options for Vanagon. They'll just be ignored when calling directory?

File: configs/projects/puppet-agent.rb (hunk 3 above)
Comment (MikaelSmith) [inline configs/projects/puppet-agent.rb]: So we need a function to generate the short version of the name, and to create the directory as before? Or going a different route for MSI staging?

File: configs/projects/puppet-agent.rb (hunk 4 above)
Comment (melissa) [inline configs/projects/puppet-agent.rb]: I put them up as an example for @McdonaldSeanp to potential take inspiration from. I think it'll cause failures as is though

File: configs/projects/puppet-agent.rb (hunk 3 above)
Comment (melissa) [inline configs/projects/puppet-agent.rb]: My solution to get the msi to create the directory with spaces was to pass in `:name` if the directory is supposed to be different than the directory we're using to build. I'm sure there are going to be a ton of consequences to this approach, but I don't know what else to do.

Comment (melissa) [inline configs/projects/puppet-agent.rb]: I don't think we can pass in the shortcuts to wix

File: configs/projects/puppet-agent.rb [hunk 5]
```diff
@@ -180,11 +206,19 @@
     proj.component "ruby-selinux"
   end
 
-  proj.directory proj.install_root
-  proj.directory proj.prefix
-  proj.directory proj.sysconfdir
+
+  #if platform.is_windows?
+  #  proj.directory proj.install_root, id: proj.company_id, name: proj.company_name
```
Comment (melissa) [inline configs/projects/puppet-agent.rb]: This hasn't been implemented yet.... I should probably take it out...

**Comment by shermdog** [comment]
:+1:

File: configs/projects/puppet-agent.rb (hunk 2 above)
Comment (jcoconnor) [inline configs/projects/puppet-agent.rb]: I think this is ok, but would prefer @jpogran to verify

**Comment by jcoconnor** [comment]
:+1: May need to check https://github.com/puppetlabs/puppet-agent/pull/523#discussion_r51808824 but we can fix this up when get light in and do all the verification.

File: configs/projects/puppet-agent.rb (hunk 2 above)
Comment (melissa) [inline configs/projects/puppet-agent.rb]: This shouldn't block this PR getting merged. We'll set the ID when the directory work is ready. It's just something to start thinking about. I also realized this isn't even the right directory for this comment!

**Pull request merged**